projects/common/src/rest/rest.service.ts

```diff
@@ -0,0 +1,45 @@
+import { Inject, Injectable, InjectionToken } from '@angular/core';
+import { Observable } from 'rxjs';
+
+import { HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
+
+interface RestOptions {
+  uri: string;
+}
+export interface IRequestOptions {
+  headers?: HttpHeaders;
+  observe?: 'body';
+  params?: HttpParams;
+  reportProgress?: boolean;
+  responseType?: 'json';
+  withCredentials?: boolean;
+  body?: object;
+}
+
+export const REST_URI = new InjectionToken<RestOptions>('REST_URI');
+
+@Injectable({
+  providedIn: 'root'
+})
+export class RestClientService {
```

Comment:
Suggestion: This service can be deleted. Apart from adding the BASE_URL, this service seems to be a wrapper on the native http service from Angular. Instead of creating this wrapper and separate tests for it, how about just providing REST_URI as a constant to import wherever http service should be used?

Example: https://github.com/razorpay/hypertrace-ui/blob/rzp_main/projects/graphql-client/src/graphql-config.ts#L6

Reply:
As discussed, maybe we can just rename this service to make it clear it only communicates to the backend user service.

Reply:
I think, thinking the way HT codebase is written, it makes sense to have a service, given we might need to have a data source linked to the service, but let's make the service more generic to have different `URLs` i.e. not have any `BASE_URL`. We can add more logic around building params and stuff later on.

Reply:
If we remove the BASE_URL from this service, then it practically becomes a duplicate of the native http service with no additional benefit. The only benefit of having this service right now is that we don't need to worry about injecting the BASE_URL token.

Reply:
Yeah, I was thinking we will need a data source to go with in case you want to create a dashboard JSON the way we have for grapqhl, so that's the reason I think this will get more stuff added to it in future.

Reply:
Let's create the service when we have the extra stuff to add? 😄
https://martinfowler.com/bliki/Yagni.html